### PR TITLE
WIP: Upgrade Semantic Ui to use Semantic-UI-Less

### PR DESCRIPTION
**Pull request by andrewleyva**
# problem statement

The build is heavy and slow and is integrated in such a way that updates could overwrite some of our code.

# solution

1. Use Semantic-UI-Less instead of Semantic-UI package to pull in CSS for Semantic to bundle with our additional components.
- Remove Semantic-UI package and folder (removing Gulp entirely)
- Add semantic-ui-less and less packages to npm
- Update Builder Script to build less with less.render and ship to Lib folder
2. Use Less-loader to replace the old watcher

# Notes:
Currently we must use Less v2.7.2 because the latest (3.0.0-alpha.3) results in build errors for semantic-ui. And issue is open: https://github.com/Semantic-Org/Semantic-UI-LESS/issues/30#issue-265910701

### CODE REVIEW

File: scripts/builder.js [hunk 1]
```diff
@@ -25,8 +26,7 @@ module.exports = {
   get coverageDir () { return path.join(this.projectRoot, 'coverage') },
   get distDir () { return path.join(this.projectRoot, 'lib') },
   get projectRoot () { return path.resolve(__dirname, '..') },
-  get semanticDist () { return path.join(this.semanticPath, 'dist') },
-  get semanticPath () { return path.join(this.projectRoot, 'semantic') },
+  get semanticPath () { return path.join(this.projectRoot, 'semantic-ui-less') },
```
Comment (cdaringe) [inline scripts/builder.js]: look into `require.resolve`

`require.resolve('semantic-ui-less')` ==> `// /Users/andrew/octagon/node_modules/semantic-ui-less/index.less`

File: scripts/builder.js [hunk 2]
```diff
@@ -87,9 +83,22 @@ module.exports = {
...
+    const assetsDest = path.join(defaultDir,'assets')
+    await mkdirp(assetsDest)
+    await copy(assetsSource, assetsDest)
+  },
   async semanticBuild () {
-    const [stdout] = await exec([this.getBin('gulp'), 'build'].join(' '), { cwd: this.semanticPath, stdio: 'inherit' })
-    console.log(stdout)
+    console.log('Building Semantic Less')
+    const lessInputSource =path.join(this.semanticPath, 'semantic.less')
+    const lessInputStream = fs.readFileSync(lessInputSource).toString()
```
Comment (cdaringe) [inline scripts/builder.js]: `await fs.readFile(...`)

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29593329.surge.sh](http://surgereview29593329.surge.sh)

File: scripts/builder.js (hunk 2 above)
Comment (andrewleyva) [reply scripts/builder.js]: done

File: scripts/builder.js (hunk 1 above)
Comment (andrewleyva) [reply scripts/builder.js]: needed to use the full path

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29595186.surge.sh](http://surgereview29595186.surge.sh)

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29596187.surge.sh](http://surgereview29596187.surge.sh)

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29596237.surge.sh](http://surgereview29596237.surge.sh)

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29596284.surge.sh](http://surgereview29596284.surge.sh)

**Comment by cdaringe** [comment]
@andrewleyva (& @ggascoigne) updates!

- i moved the tmp build dir to a true sys-wise tmp dir
- given your CSS build is pretty fast and because i parallelized a lot of the steps, i was able to drop `less-loader`, and just do a full CSS rebuild on every change from `src/semantic-ui-theme`, vs watching the temp dir less file.  if it becomes problematic, no worries, let's port it back.  it honestly is probably a little slower, _but_, it's simpler.  "changed theme? then new theme dumped into lib"

take a look at the static styleguide from @wa11-e (yours, or mine).

- it looks like the icon on the sortable table isn't rendering.
- if you run `npm install && npm test` in the latest branch, there are some actual diffs im observing too.  not much! just some smallies.  we need to loosen the image diff threshold in the test :$
  - button padding-left|right maybe have both increased?  we should see what's up w/ that.

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29667143.surge.sh](http://surgereview29667143.surge.sh)

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29675775.surge.sh](http://surgereview29675775.surge.sh)

**Comment by andrewleyva** [comment]
Addresses issue #119.

**Comment by ggascoigne** [comment]
I'm curious what the anticipated impact on TOC and Connect v4 will be with this change.

**Comment by andrewleyva** [comment]
@cdaringe and I ran Backstop and investigated any differences. This really just cleans up the build on the Octagon side. I do have plans to address the Bundle size, issue #120, which should have more of an impact.

**Comment by ggascoigne** [comment]
Cool, I'll pull it into TOC when I return and run backstop there :)  That'll be the real test :grin:

**Pull request merged**